add(epp): aws direct connect (#2252)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/cloud-aws-directconnect.md

---
id: cloud-aws-directconnect
title: Amazon Direct Connect
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack Assets

### Templates

> The Centreon Pack **Amazon Direct Connect** brings a host template:

Reviewer: ```suggestion
The Centreon Monitoring Connector **Amazon Direct Connect** brings a host template:
```

* Cloud-Aws-Directconnect-custom-custom

It brings the following service templates:

| Service Alias      | Service Template                               | Service Description      | Default | Discovery |
|:-------------------|:-----------------------------------------------|:-------------------------|:--------|:----------|
| Virtual-Interfaces | Cloud-Aws-Directconnect-Virtual-Interfaces-Api | Check virtual interfaces | X       | X         |
| Connections        | Cloud-Aws-Directconnect-Connections-Api        | Check connections        | X       |           |

> Default services are automatically created when the host template is applied.
>
> Services templates with the *Discovery* field are linked to a service discovery rule.

### Discovery rules

> The pack provides a discovery rule to automatically discover Amazon Direct Connect resources:

Reviewer: ```suggestion
The pack provides a discovery rule to automatically discover **Amazon Direct Connect** resources:
```

> The pack provides a discovery rule to automatically discover **Amazon Direct Connect** resources:

Reviewer 2: ```suggestion
The pack provides a discovery rule to automatically discover **Amazon Direct Connect** resources.
```

> ![image](../../../assets/integrations/plugin-packs/procedures/cloud-aws-directconnect-provider.png)

Reviewer: Image file is missing.

> ![image](../../../assets/integrations/plugin-packs/procedures/cloud-aws-directconnect-provider.png)

Reviewer 2: ```suggestion
```

| directconnect:describe-connections        | Returns all connections.                             |
| directconnect:describe-virtual-interfaces | Returns all virtual interfaces.                      |
| cloudwatch:getMetricStatistics            | Get metrics from the AWS/DX namespace on Cloudwatch. |

### Plugin dependencies

To interact with Amazon APIs, you can use either use the *awscli* binary provided by Amazon or *paws*, a Perl AWS SDK (recommended). You must install it on every poller expected to monitor AWS resources.

> For now, it is not possible to use *paws* if you are using a proxy to reach AWS Cloudwatch APIs.

<Tabs groupId="sync">
<TabItem value="perl-Paws-installation" label="perl-Paws-installation">

```bashn
yum install perl-Paws
```

</TabItem>
<TabItem value="aws-cli-installation" label="aws-cli-installation">

```bash
curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
unzip awscliv2.zip
sudo ./aws/install
```

</TabItem>
</Tabs>

## Setup

### Monitoring Pack

If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the pack displayed within the

> **Configuration > Plugin Packs > Manager** menu.

Reviewer: ```suggestion
**Configuration > Monitoring Connectors Manager** menu.
```

If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-cloud-aws-directconnect
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-cloud-aws-directconnect
```

</TabItem>
<TabItem value="Debian 11" label="Debian 11">

```bash
apt install centreon-pack-cloud-aws-directconnect
```

</TabItem>
</Tabs>

Whatever the license type (*online* or *offline*), install the **Amazon Direct Connect** Pack through

> the **Configuration > Plugin Packs > Manager** menu.

Reviewer: ```suggestion
the **Configuration > Monitoring Connectors Manager** menu.
```

### Plugin

Since Centreon 22.04, you can benefit from the 'Automatic plugin installation' feature.
When this feature is enabled, you can skip the installation part below.

You still have to manually install the plugin on the poller(s) when:
- Automatic plugin installation is turned off
- You want to run a discovery job from a poller that doesn't monitor any resource of this kind yet

> More information in the [Installing the plugin](/docs/monitoring/pluginpacks/#installing-the-plugin) section.

Use the commands below according to your operating system's package manager:

> 

Reviewer: Is EL9 required too?

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-plugin-Cloud-Aws-Directconnect-Api
```

> </TabItem>

Reviewer 2: ```suggestion
</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-plugin-Cloud-Aws-Directconnect-Api
```

</TabItem>
```

```

### Available custom modes

All available custom modes can be displayed by adding the `--list-custommode` parameter to
the command:

```bash
/usr/lib/centreon/plugins/centreon_aws_directconnect_api.pl \
	--plugin=cloud::aws::directconnect::plugin \
    --list-custommode
```

The plugin brings the following custom modes:

* paws
* awscli

### Available modes

All available modes can be displayed by adding the `--list-mode` parameter to
the command:

```bash
/usr/lib/centreon/plugins/centreon_aws_directconnect_api.pl \
	--plugin=cloud::aws::directconnect::plugin \
    --list-mode
```

The plugin brings the following modes:

| Mode                    | Template                                       |
|:------------------------|:-----------------------------------------------|
| connections             | Cloud-Aws-Directconnect-Connections-Api        |
| discovery               | Used for host discovery                        |
| list-connections        | Not used in this Plugin Pack                   |
| list-virtual-interfaces | Used for service discovery                     |
| virtual-interfaces      | Cloud-Aws-Directconnect-Virtual-Interfaces-Api |

### Available options

> #### Global optionsAll global options are listed here:

Reviewer: ```suggestion
#### Global options

All global options are listed here:
```
